File: powerapi-powermeter/AppMonitorSigarScala/src/main/scala/AppMonitorExample.scala
import org.powerapi.PowerMeter
import org.powerapi.module.cpu.simple.SigarCpuSimpleModule
import org.powerapi.core.target._
import org.powerapi.reporter.JFreeChartDisplay
import scala.concurrent.duration._

object Monitor extends App {
  val cpu = PowerMeter.loadModule(SigarCpuSimpleModule())
  val chart = new JFreeChartDisplay
  val monitoring = cpu.monitor(1.second)(args(0)) to chart

  cpu.waitFor(5.minutes)
  
  monitoring.cancel
  cpu.shutdown
}

File: powerapi-powermeter/AppMonitorSigarJava/src/main/java/AppMonitorExample.java
import org.powerapi.PowerMeter;
import org.powerapi.PowerModule;
import org.powerapi.PowerMonitoring;
import org.powerapi.module.cpu.simple.SigarCpuSimpleModule;
import org.powerapi.reporter.JFreeChartDisplay;
import org.powerapi.core.target.Target;
import org.powerapi.core.target.Application;
import scala.concurrent.duration.Duration;
import scala.collection.JavaConversions;
import java.util.ArrayList;
import java.util.Arrays;

public class AppMonitorExample {
  public static void main (String[] args) {
    ArrayList<PowerModule> modules = new ArrayList<PowerModule>(Arrays.asList(SigarCpuSimpleModule.apply()));
    PowerMeter cpu_simple = PowerMeter.loadModule(JavaConversions.asScalaBuffer(modules));
    JFreeChartDisplay chart = new JFreeChartDisplay();
    ArrayList<Target> targets = new ArrayList<Target>(Arrays.asList(new Application(args[0])));
    PowerMonitoring monitoring = cpu_simple.monitor(Duration.create(1L, "seconds"), JavaConversions.asScalaBuffer(targets));
    monitoring.to(chart);
    
    cpu_simple.waitFor(Duration.create(5, "minutes"));
    
    monitoring.cancel();
    cpu_simple.shutdown();
  }
}

